externalpolicy/property_list_test.go

```diff
@@ -51,6 +52,46 @@ func Test_PropertyList_compatible(t *testing.T) {
 			}
 		}
 	}
+
+	p1 = `[{"name":"prop1","value":"val1","type":"string"}]`
+	p2 = `[{"name":"prop1","value":"val1"}]`
+	if pl1 = create_PropertyList(p1, t); pl1 != nil {
+		if pl2 = create_PropertyList(p2, t); pl2 != nil {
+			if err := pl1.Compatible_With(pl2); err != nil {
+				t.Errorf("Error: %v is compatible with %v, error was %v\n", p1, p2, err)
+			}
+		}
+	}
+
+	p1 = `[{"name":"prop1","value":"12"},{"name":"prop2","value":"1.0.3","type":"version"}]`
+	p2 = `[{"name":"prop1","value":12,"type":"float"},{"name":"prop3","value":"val4"}]`
+	if pl1 = create_PropertyList(p1, t); pl1 != nil {
+		if pl2 = create_PropertyList(p2, t); pl2 != nil {
+			if err := pl1.Compatible_With(pl2); err != nil {
+				t.Errorf("Error: %v is compatible with %v, error was %v\n", p1, p2, err)
+			}
+		}
+	}
+
+	p1 = `[{"name":"prop1","value":"1.2.53"},{"name":"prop2","value":"val2"}]`
+	p2 = `[{"name":"prop2","value":"val2"},{"name":"prop1","value":"1.2.53","type":"version"}]`
+	if pl1 = create_PropertyList(p1, t); pl1 != nil {
+		if pl2 = create_PropertyList(p2, t); pl2 != nil {
+			if err := pl1.Compatible_With(pl2); err != nil {
+				t.Errorf("Error: %v is compatible with %v, error was %v\n", p1, p2, err)
+			}
+		}
+	}
+
+	p1 = `[{"name":"prop1","value":["a","b","c"],"type":"list of string"},{"name":"prop2","value":"val2"}]`
```

Comment:
list of string is actually supposed to be comma separated values in 1 string, not a proper json array